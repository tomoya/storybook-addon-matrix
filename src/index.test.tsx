import * as React from 'react';
import { mount } from 'enzyme';
import { withMatrix } from '.';

type BaseComponentProps = {
  title?: string;
  text: string;
};

const BaseComponent: React.FC<BaseComponentProps> = ({ title, text }) => (
  <div>
    {title && <h1>{title}</h1>}
    <p>{text}</p>
  </div>
);

const TestComponent: React.FC = () => <BaseComponent title="TITLE" text="TEXT" />;

describe('addon Matrix', () => {
  const createStoryFn = (Component: React.ElementType) => () => <Component />;
  const storyFn = createStoryFn(TestComponent);
  it('should render <Matrix />', () => {
    const Matrix = withMatrix()(storyFn, {
      parameters: {
        component: BaseComponent,
        matrix: {
          pattern: {
            title: [undefined, 'title pattern 1'],
            text: ['text pattern 1', 'text pattern 2'],
          },
        },
      },
    });
    expect(mount(Matrix)).toMatchSnapshot();
  });

  it('should render <Matrix /> with backgroundColor', () => {
    const Matrix = withMatrix()(storyFn, {
      parameters: {
        component: BaseComponent,
        matrix: {
          pattern: {
            title: ['title with background'],
          },
          backgroundColor: 'black',
        },
      },
    });
    expect(mount(Matrix)).toMatchSnapshot();
  });

  it('should render <Matrix /> with showOriginalL true', () => {
    const Matrix = withMatrix()(storyFn, {
      parameters: {
        component: BaseComponent,
        matrix: {
          pattern: {
            title: ['title with showOriginal'],
          },
          showOriginal: true,
        },
      },
    });
    expect(mount(Matrix)).toMatchSnapshot();
  });

  it('should render <Error /> when component is not exist', () => {
    const Matrix = withMatrix()(storyFn, {
      parameters: {
        matrix: {
          pattern: {
            title: [undefined, 'title pattern 1'],
            text: ['text pattern 1', 'text pattern 2'],
          },
        },
      },
    });
    expect(mount(Matrix)).toMatchSnapshot();
  });

  it('should render <Error /> when pattern is not exist', () => {
    const Matrix = withMatrix()(storyFn, {
      parameters: {
        component: BaseComponent,
        matrix: {},
      },
    });
    expect(mount(Matrix)).toMatchSnapshot();
  });

  it('should render <Error /> when component and pattern are not exist', () => {
    const Matrix = withMatrix()(storyFn, {
      parameters: {
        matrix: {},
      },
    });
    expect(mount(Matrix)).toMatchSnapshot();
  });
});
