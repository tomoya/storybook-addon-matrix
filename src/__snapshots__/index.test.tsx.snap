// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addon Matrix should render <Error /> when component and pattern are not exist 1`] = `
<Component
  messages={
    Array [
      "Default export component must be present",
      "Parameter pattern must be present",
    ]
  }
>
  <Styled(div)
    bg="red"
    color="white"
    p={2}
  >
    <div
      className="css-r76ibh"
      color="white"
    >
      <Styled(span)
        color="rgb(51, 51, 51)"
        fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
        fontSize={2}
      >
        <span
          className="css-16rypc2"
          color="rgb(51, 51, 51)"
          fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
          fontSize={2}
        >
          Storybook Addon Matrix Error
        </span>
      </Styled(span)>
      <ul>
        <li
          key="Default export component must be present"
        >
          <Styled(span)
            color="rgb(51, 51, 51)"
            fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
          >
            <span
              className="css-hj85ni"
              color="rgb(51, 51, 51)"
              fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
            >
              Default export component must be present
            </span>
          </Styled(span)>
        </li>
        <li
          key="Parameter pattern must be present"
        >
          <Styled(span)
            color="rgb(51, 51, 51)"
            fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
          >
            <span
              className="css-hj85ni"
              color="rgb(51, 51, 51)"
              fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
            >
              Parameter pattern must be present
            </span>
          </Styled(span)>
        </li>
      </ul>
    </div>
  </Styled(div)>
</Component>
`;

exports[`addon Matrix should render <Error /> when component is not exist 1`] = `
<Component
  messages={
    Array [
      "Default export component must be present",
    ]
  }
>
  <Styled(div)
    bg="red"
    color="white"
    p={2}
  >
    <div
      className="css-r76ibh"
      color="white"
    >
      <Styled(span)
        color="rgb(51, 51, 51)"
        fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
        fontSize={2}
      >
        <span
          className="css-16rypc2"
          color="rgb(51, 51, 51)"
          fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
          fontSize={2}
        >
          Storybook Addon Matrix Error
        </span>
      </Styled(span)>
      <ul>
        <li
          key="Default export component must be present"
        >
          <Styled(span)
            color="rgb(51, 51, 51)"
            fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
          >
            <span
              className="css-hj85ni"
              color="rgb(51, 51, 51)"
              fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
            >
              Default export component must be present
            </span>
          </Styled(span)>
        </li>
      </ul>
    </div>
  </Styled(div)>
</Component>
`;

exports[`addon Matrix should render <Error /> when pattern is not exist 1`] = `
<Component
  messages={
    Array [
      "Parameter pattern must be present",
    ]
  }
>
  <Styled(div)
    bg="red"
    color="white"
    p={2}
  >
    <div
      className="css-r76ibh"
      color="white"
    >
      <Styled(span)
        color="rgb(51, 51, 51)"
        fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
        fontSize={2}
      >
        <span
          className="css-16rypc2"
          color="rgb(51, 51, 51)"
          fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
          fontSize={2}
        >
          Storybook Addon Matrix Error
        </span>
      </Styled(span)>
      <ul>
        <li
          key="Parameter pattern must be present"
        >
          <Styled(span)
            color="rgb(51, 51, 51)"
            fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
          >
            <span
              className="css-hj85ni"
              color="rgb(51, 51, 51)"
              fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
            >
              Parameter pattern must be present
            </span>
          </Styled(span)>
        </li>
      </ul>
    </div>
  </Styled(div)>
</Component>
`;

exports[`addon Matrix should render <Matrix /> 1`] = `
Array [
  <Styled(div)
    mb={3}
  >
    <div
      className="css-yoonho"
    >
      <TestComponent>
        <BaseComponent
          text="TEXT"
          title="TITLE"
        >
          <div>
            <h1>
              TITLE
            </h1>
            <p>
              TEXT
            </p>
          </div>
        </BaseComponent>
      </TestComponent>
    </div>
  </Styled(div)>,
  <Component
    component={[Function]}
    originalProps={Object {}}
    propsPattern={
      Object {
        "text": Array [
          "text pattern 1",
          "text pattern 2",
        ],
        "title": Array [
          undefined,
          "title pattern 1",
        ],
      }
    }
  >
    <Styled(div)
      display="flex"
      flexWrap="wrap"
      justifyContent="space-around"
    >
      <div
        className="css-bedbvn"
        display="flex"
      >
        <Styled(div)
          borderRadius={4}
          boxShadow="rgba(0, 0, 0, 0.1) 0px 1px 3px 1px, rgba(0, 0, 0, 0.067) 0px 0px 0px 1px"
          display="flex"
          id="%7B%22text%22%3A%22text%20pattern%201%22%7D--1"
          key="%7B%22text%22%3A%22text%20pattern%201%22%7D"
          mb={3}
          position="relative"
          width={0.3}
        >
          <div
            className="css-srvi11"
            display="flex"
            id="%7B%22text%22%3A%22text%20pattern%201%22%7D--1"
            width={0.3}
          >
            <Styled(div)
              bg="rgb(246, 249, 252)"
              p={2}
              width={0.3333333333333333}
            >
              <div
                className="css-15pd0l3"
                width={0.3333333333333333}
              >
                <Styled(div)
                  key="title"
                >
                  <div
                    className="css-1qi6ewn"
                  >
                    <Styled(div)
                      mb={2}
                    >
                      <div
                        className="css-ow3zxq"
                      >
                        <Styled(span)
                          color="rgba(51, 51, 51, 0.75)"
                          fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
                          fontWeight="bold"
                        >
                          <span
                            className="css-19noksk"
                            color="rgba(51, 51, 51, 0.75)"
                            fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
                            fontWeight="bold"
                          >
                            title
                          </span>
                        </Styled(span)>
                        :
                         
                        <Styled(span)
                          color="rgba(51, 51, 51, 0.25)"
                          fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
                        >
                          <span
                            className="css-12p2gsl"
                            color="rgba(51, 51, 51, 0.25)"
                            fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
                          >
                            undefined
                          </span>
                        </Styled(span)>
                      </div>
                    </Styled(div)>
                  </div>
                </Styled(div)>
                <Styled(div)
                  key="text"
                >
                  <div
                    className="css-1qi6ewn"
                  >
                    <Styled(div)
                      mb={2}
                    >
                      <div
                        className="css-ow3zxq"
                      >
                        <Styled(span)
                          color="rgba(51, 51, 51, 0.75)"
                          fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
                          fontWeight="bold"
                        >
                          <span
                            className="css-19noksk"
                            color="rgba(51, 51, 51, 0.75)"
                            fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
                            fontWeight="bold"
                          >
                            text
                          </span>
                        </Styled(span)>
                        :
                         
                        <Styled(span)
                          color="rgba(51, 51, 51, 0.75)"
                          fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
                        >
                          <span
                            className="css-8v37fk"
                            color="rgba(51, 51, 51, 0.75)"
                            fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
                          >
                            text pattern 1
                          </span>
                        </Styled(span)>
                      </div>
                    </Styled(div)>
                  </div>
                </Styled(div)>
              </div>
            </Styled(div)>
            <Styled(div)
              overflowX="scroll"
              p={2}
              pb={4}
              width={0.6666666666666666}
            >
              <div
                className="css-p4w7s8"
                width={0.6666666666666666}
              >
                <BaseComponent
                  text="text pattern 1"
                >
                  <div>
                    <p>
                      text pattern 1
                    </p>
                  </div>
                </BaseComponent>
              </div>
            </Styled(div)>
            <a
              href="#%7B%22text%22%3A%22text%20pattern%201%22%7D--1"
            >
              <Styled(div)
                bg="rgba(255,255,255,0.6)"
                borderBottomRightRadius={4}
                borderTopLeftRadius={2}
                bottom="0"
                pb={1}
                pl={2}
                position="absolute"
                pr={2}
                pt={1}
                right="0"
              >
                <div
                  className="css-1nrfhbr"
                >
                  <Styled(span)
                    color="rgba(51, 51, 51, 0.6)"
                    fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
                    fontSize={1}
                  >
                    <span
                      className="css-g3n3x7"
                      color="rgba(51, 51, 51, 0.6)"
                      fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
                      fontSize={1}
                    >
                      1
                       / 
                      4
                    </span>
                  </Styled(span)>
                </div>
              </Styled(div)>
            </a>
          </div>
        </Styled(div)>
        <Styled(div)
          borderRadius={4}
          boxShadow="rgba(0, 0, 0, 0.1) 0px 1px 3px 1px, rgba(0, 0, 0, 0.067) 0px 0px 0px 1px"
          display="flex"
          id="%7B%22title%22%3A%22title%20pattern%201%22%2C%22text%22%3A%22text%20pattern%201%22%7D--2"
          key="%7B%22title%22%3A%22title%20pattern%201%22%2C%22text%22%3A%22text%20pattern%201%22%7D"
          mb={3}
          position="relative"
          width={0.3}
        >
          <div
            className="css-srvi11"
            display="flex"
            id="%7B%22title%22%3A%22title%20pattern%201%22%2C%22text%22%3A%22text%20pattern%201%22%7D--2"
            width={0.3}
          >
            <Styled(div)
              bg="rgb(246, 249, 252)"
              p={2}
              width={0.3333333333333333}
            >
              <div
                className="css-15pd0l3"
                width={0.3333333333333333}
              >
                <Styled(div)
                  key="title"
                >
                  <div
                    className="css-1qi6ewn"
                  >
                    <Styled(div)
                      mb={2}
                    >
                      <div
                        className="css-ow3zxq"
                      >
                        <Styled(span)
                          color="rgba(51, 51, 51, 0.75)"
                          fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
                          fontWeight="bold"
                        >
                          <span
                            className="css-19noksk"
                            color="rgba(51, 51, 51, 0.75)"
                            fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
                            fontWeight="bold"
                          >
                            title
                          </span>
                        </Styled(span)>
                        :
                         
                        <Styled(span)
                          color="rgba(51, 51, 51, 0.75)"
                          fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
                        >
                          <span
                            className="css-8v37fk"
                            color="rgba(51, 51, 51, 0.75)"
                            fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
                          >
                            title pattern 1
                          </span>
                        </Styled(span)>
                      </div>
                    </Styled(div)>
                  </div>
                </Styled(div)>
                <Styled(div)
                  key="text"
                >
                  <div
                    className="css-1qi6ewn"
                  >
                    <Styled(div)
                      mb={2}
                    >
                      <div
                        className="css-ow3zxq"
                      >
                        <Styled(span)
                          color="rgba(51, 51, 51, 0.75)"
                          fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
                          fontWeight="bold"
                        >
                          <span
                            className="css-19noksk"
                            color="rgba(51, 51, 51, 0.75)"
                            fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
                            fontWeight="bold"
                          >
                            text
                          </span>
                        </Styled(span)>
                        :
                         
                        <Styled(span)
                          color="rgba(51, 51, 51, 0.75)"
                          fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
                        >
                          <span
                            className="css-8v37fk"
                            color="rgba(51, 51, 51, 0.75)"
                            fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
                          >
                            text pattern 1
                          </span>
                        </Styled(span)>
                      </div>
                    </Styled(div)>
                  </div>
                </Styled(div)>
              </div>
            </Styled(div)>
            <Styled(div)
              overflowX="scroll"
              p={2}
              pb={4}
              width={0.6666666666666666}
            >
              <div
                className="css-p4w7s8"
                width={0.6666666666666666}
              >
                <BaseComponent
                  text="text pattern 1"
                  title="title pattern 1"
                >
                  <div>
                    <h1>
                      title pattern 1
                    </h1>
                    <p>
                      text pattern 1
                    </p>
                  </div>
                </BaseComponent>
              </div>
            </Styled(div)>
            <a
              href="#%7B%22title%22%3A%22title%20pattern%201%22%2C%22text%22%3A%22text%20pattern%201%22%7D--2"
            >
              <Styled(div)
                bg="rgba(255,255,255,0.6)"
                borderBottomRightRadius={4}
                borderTopLeftRadius={2}
                bottom="0"
                pb={1}
                pl={2}
                position="absolute"
                pr={2}
                pt={1}
                right="0"
              >
                <div
                  className="css-1nrfhbr"
                >
                  <Styled(span)
                    color="rgba(51, 51, 51, 0.6)"
                    fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
                    fontSize={1}
                  >
                    <span
                      className="css-g3n3x7"
                      color="rgba(51, 51, 51, 0.6)"
                      fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
                      fontSize={1}
                    >
                      2
                       / 
                      4
                    </span>
                  </Styled(span)>
                </div>
              </Styled(div)>
            </a>
          </div>
        </Styled(div)>
        <Styled(div)
          borderRadius={4}
          boxShadow="rgba(0, 0, 0, 0.1) 0px 1px 3px 1px, rgba(0, 0, 0, 0.067) 0px 0px 0px 1px"
          display="flex"
          id="%7B%22text%22%3A%22text%20pattern%202%22%7D--3"
          key="%7B%22text%22%3A%22text%20pattern%202%22%7D"
          mb={3}
          position="relative"
          width={0.3}
        >
          <div
            className="css-srvi11"
            display="flex"
            id="%7B%22text%22%3A%22text%20pattern%202%22%7D--3"
            width={0.3}
          >
            <Styled(div)
              bg="rgb(246, 249, 252)"
              p={2}
              width={0.3333333333333333}
            >
              <div
                className="css-15pd0l3"
                width={0.3333333333333333}
              >
                <Styled(div)
                  key="title"
                >
                  <div
                    className="css-1qi6ewn"
                  >
                    <Styled(div)
                      mb={2}
                    >
                      <div
                        className="css-ow3zxq"
                      >
                        <Styled(span)
                          color="rgba(51, 51, 51, 0.75)"
                          fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
                          fontWeight="bold"
                        >
                          <span
                            className="css-19noksk"
                            color="rgba(51, 51, 51, 0.75)"
                            fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
                            fontWeight="bold"
                          >
                            title
                          </span>
                        </Styled(span)>
                        :
                         
                        <Styled(span)
                          color="rgba(51, 51, 51, 0.25)"
                          fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
                        >
                          <span
                            className="css-12p2gsl"
                            color="rgba(51, 51, 51, 0.25)"
                            fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
                          >
                            undefined
                          </span>
                        </Styled(span)>
                      </div>
                    </Styled(div)>
                  </div>
                </Styled(div)>
                <Styled(div)
                  key="text"
                >
                  <div
                    className="css-1qi6ewn"
                  >
                    <Styled(div)
                      mb={2}
                    >
                      <div
                        className="css-ow3zxq"
                      >
                        <Styled(span)
                          color="rgba(51, 51, 51, 0.75)"
                          fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
                          fontWeight="bold"
                        >
                          <span
                            className="css-19noksk"
                            color="rgba(51, 51, 51, 0.75)"
                            fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
                            fontWeight="bold"
                          >
                            text
                          </span>
                        </Styled(span)>
                        :
                         
                        <Styled(span)
                          color="rgba(51, 51, 51, 0.75)"
                          fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
                        >
                          <span
                            className="css-8v37fk"
                            color="rgba(51, 51, 51, 0.75)"
                            fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
                          >
                            text pattern 2
                          </span>
                        </Styled(span)>
                      </div>
                    </Styled(div)>
                  </div>
                </Styled(div)>
              </div>
            </Styled(div)>
            <Styled(div)
              overflowX="scroll"
              p={2}
              pb={4}
              width={0.6666666666666666}
            >
              <div
                className="css-p4w7s8"
                width={0.6666666666666666}
              >
                <BaseComponent
                  text="text pattern 2"
                >
                  <div>
                    <p>
                      text pattern 2
                    </p>
                  </div>
                </BaseComponent>
              </div>
            </Styled(div)>
            <a
              href="#%7B%22text%22%3A%22text%20pattern%202%22%7D--3"
            >
              <Styled(div)
                bg="rgba(255,255,255,0.6)"
                borderBottomRightRadius={4}
                borderTopLeftRadius={2}
                bottom="0"
                pb={1}
                pl={2}
                position="absolute"
                pr={2}
                pt={1}
                right="0"
              >
                <div
                  className="css-1nrfhbr"
                >
                  <Styled(span)
                    color="rgba(51, 51, 51, 0.6)"
                    fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
                    fontSize={1}
                  >
                    <span
                      className="css-g3n3x7"
                      color="rgba(51, 51, 51, 0.6)"
                      fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
                      fontSize={1}
                    >
                      3
                       / 
                      4
                    </span>
                  </Styled(span)>
                </div>
              </Styled(div)>
            </a>
          </div>
        </Styled(div)>
        <Styled(div)
          borderRadius={4}
          boxShadow="rgba(0, 0, 0, 0.1) 0px 1px 3px 1px, rgba(0, 0, 0, 0.067) 0px 0px 0px 1px"
          display="flex"
          id="%7B%22title%22%3A%22title%20pattern%201%22%2C%22text%22%3A%22text%20pattern%202%22%7D--4"
          key="%7B%22title%22%3A%22title%20pattern%201%22%2C%22text%22%3A%22text%20pattern%202%22%7D"
          mb={3}
          position="relative"
          width={0.3}
        >
          <div
            className="css-srvi11"
            display="flex"
            id="%7B%22title%22%3A%22title%20pattern%201%22%2C%22text%22%3A%22text%20pattern%202%22%7D--4"
            width={0.3}
          >
            <Styled(div)
              bg="rgb(246, 249, 252)"
              p={2}
              width={0.3333333333333333}
            >
              <div
                className="css-15pd0l3"
                width={0.3333333333333333}
              >
                <Styled(div)
                  key="title"
                >
                  <div
                    className="css-1qi6ewn"
                  >
                    <Styled(div)
                      mb={2}
                    >
                      <div
                        className="css-ow3zxq"
                      >
                        <Styled(span)
                          color="rgba(51, 51, 51, 0.75)"
                          fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
                          fontWeight="bold"
                        >
                          <span
                            className="css-19noksk"
                            color="rgba(51, 51, 51, 0.75)"
                            fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
                            fontWeight="bold"
                          >
                            title
                          </span>
                        </Styled(span)>
                        :
                         
                        <Styled(span)
                          color="rgba(51, 51, 51, 0.75)"
                          fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
                        >
                          <span
                            className="css-8v37fk"
                            color="rgba(51, 51, 51, 0.75)"
                            fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
                          >
                            title pattern 1
                          </span>
                        </Styled(span)>
                      </div>
                    </Styled(div)>
                  </div>
                </Styled(div)>
                <Styled(div)
                  key="text"
                >
                  <div
                    className="css-1qi6ewn"
                  >
                    <Styled(div)
                      mb={2}
                    >
                      <div
                        className="css-ow3zxq"
                      >
                        <Styled(span)
                          color="rgba(51, 51, 51, 0.75)"
                          fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
                          fontWeight="bold"
                        >
                          <span
                            className="css-19noksk"
                            color="rgba(51, 51, 51, 0.75)"
                            fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
                            fontWeight="bold"
                          >
                            text
                          </span>
                        </Styled(span)>
                        :
                         
                        <Styled(span)
                          color="rgba(51, 51, 51, 0.75)"
                          fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
                        >
                          <span
                            className="css-8v37fk"
                            color="rgba(51, 51, 51, 0.75)"
                            fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
                          >
                            text pattern 2
                          </span>
                        </Styled(span)>
                      </div>
                    </Styled(div)>
                  </div>
                </Styled(div)>
              </div>
            </Styled(div)>
            <Styled(div)
              overflowX="scroll"
              p={2}
              pb={4}
              width={0.6666666666666666}
            >
              <div
                className="css-p4w7s8"
                width={0.6666666666666666}
              >
                <BaseComponent
                  text="text pattern 2"
                  title="title pattern 1"
                >
                  <div>
                    <h1>
                      title pattern 1
                    </h1>
                    <p>
                      text pattern 2
                    </p>
                  </div>
                </BaseComponent>
              </div>
            </Styled(div)>
            <a
              href="#%7B%22title%22%3A%22title%20pattern%201%22%2C%22text%22%3A%22text%20pattern%202%22%7D--4"
            >
              <Styled(div)
                bg="rgba(255,255,255,0.6)"
                borderBottomRightRadius={4}
                borderTopLeftRadius={2}
                bottom="0"
                pb={1}
                pl={2}
                position="absolute"
                pr={2}
                pt={1}
                right="0"
              >
                <div
                  className="css-1nrfhbr"
                >
                  <Styled(span)
                    color="rgba(51, 51, 51, 0.6)"
                    fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
                    fontSize={1}
                  >
                    <span
                      className="css-g3n3x7"
                      color="rgba(51, 51, 51, 0.6)"
                      fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
                      fontSize={1}
                    >
                      4
                       / 
                      4
                    </span>
                  </Styled(span)>
                </div>
              </Styled(div)>
            </a>
          </div>
        </Styled(div)>
      </div>
    </Styled(div)>
  </Component>,
]
`;

exports[`addon Matrix should render <Matrix /> with backgroundColor 1`] = `
Array [
  <Styled(div)
    mb={3}
  >
    <div
      className="css-yoonho"
    >
      <TestComponent>
        <BaseComponent
          text="TEXT"
          title="TITLE"
        >
          <div>
            <h1>
              TITLE
            </h1>
            <p>
              TEXT
            </p>
          </div>
        </BaseComponent>
      </TestComponent>
    </div>
  </Styled(div)>,
  <Component
    backgroundColor="black"
    component={[Function]}
    originalProps={Object {}}
    propsPattern={
      Object {
        "title": Array [
          "title with background",
        ],
      }
    }
  >
    <Styled(div)
      display="flex"
      flexWrap="wrap"
      justifyContent="space-around"
    >
      <div
        className="css-bedbvn"
        display="flex"
      >
        <Styled(div)
          backgroundColor="black"
          borderRadius={4}
          boxShadow="rgba(0, 0, 0, 0.1) 0px 1px 3px 1px, rgba(0, 0, 0, 0.067) 0px 0px 0px 1px"
          display="flex"
          id="%7B%22title%22%3A%22title%20with%20background%22%7D--1"
          key="%7B%22title%22%3A%22title%20with%20background%22%7D"
          mb={3}
          position="relative"
          width={0.3}
        >
          <div
            className="css-1m3w2x9"
            display="flex"
            id="%7B%22title%22%3A%22title%20with%20background%22%7D--1"
            width={0.3}
          >
            <Styled(div)
              bg="rgb(246, 249, 252)"
              p={2}
              width={0.3333333333333333}
            >
              <div
                className="css-15pd0l3"
                width={0.3333333333333333}
              >
                <Styled(div)
                  key="title"
                >
                  <div
                    className="css-1qi6ewn"
                  >
                    <Styled(div)
                      mb={2}
                    >
                      <div
                        className="css-ow3zxq"
                      >
                        <Styled(span)
                          color="rgba(51, 51, 51, 0.75)"
                          fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
                          fontWeight="bold"
                        >
                          <span
                            className="css-19noksk"
                            color="rgba(51, 51, 51, 0.75)"
                            fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
                            fontWeight="bold"
                          >
                            title
                          </span>
                        </Styled(span)>
                        :
                         
                        <Styled(span)
                          color="rgba(51, 51, 51, 0.75)"
                          fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
                        >
                          <span
                            className="css-8v37fk"
                            color="rgba(51, 51, 51, 0.75)"
                            fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
                          >
                            title with background
                          </span>
                        </Styled(span)>
                      </div>
                    </Styled(div)>
                  </div>
                </Styled(div)>
              </div>
            </Styled(div)>
            <Styled(div)
              overflowX="scroll"
              p={2}
              pb={4}
              width={0.6666666666666666}
            >
              <div
                className="css-p4w7s8"
                width={0.6666666666666666}
              >
                <BaseComponent
                  title="title with background"
                >
                  <div>
                    <h1>
                      title with background
                    </h1>
                    <p />
                  </div>
                </BaseComponent>
              </div>
            </Styled(div)>
            <a
              href="#%7B%22title%22%3A%22title%20with%20background%22%7D--1"
            >
              <Styled(div)
                bg="rgba(255,255,255,0.6)"
                borderBottomRightRadius={4}
                borderTopLeftRadius={2}
                bottom="0"
                pb={1}
                pl={2}
                position="absolute"
                pr={2}
                pt={1}
                right="0"
              >
                <div
                  className="css-1nrfhbr"
                >
                  <Styled(span)
                    color="rgba(51, 51, 51, 0.6)"
                    fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
                    fontSize={1}
                  >
                    <span
                      className="css-g3n3x7"
                      color="rgba(51, 51, 51, 0.6)"
                      fontFamily="\\"Nunito Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif"
                      fontSize={1}
                    >
                      1
                       / 
                      1
                    </span>
                  </Styled(span)>
                </div>
              </Styled(div)>
            </a>
          </div>
        </Styled(div)>
      </div>
    </Styled(div)>
  </Component>,
]
`;
